app/domain/operations/transmittable/create_job.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Operations
+  module Transmittable
+    # create job operation that takes params of key (required), started_at(required), publish_on(required)
+    class CreateJob
+      include Dry::Monads[:result, :do, :try]
+
+      def call(params)
+        values = yield validate_params(params)
+        job_hash = yield create_job_hash(values)
+        job_entity = yield create_job_entity(job_hash)
+        job = yield create_job(job_entity)
+        Success(job)
+      end
+
+      private
+
+      def validate_params(params)
+        return Failure('Cannot create a job without a key as a symbol') unless params[:key].is_a?(Symbol)
+        return Failure('Cannot create a job without a started_at as a DateTime') unless params[:started_at].is_a?(DateTime)
+        return Failure('Cannot create a job without a publish_on as a DateTime') unless params[:publish_on].is_a?(DateTime)
+
+        Success(params)
+      end
+
+      def create_job_hash(values)
+        Success({
+                  job_id: generate_job_id(values[:key]),
+                  saga_id: values[:saga_id],
+                  key: values[:key],
+                  title: values[:title],
+                  description: values[:description],
+                  publish_on: values[:publish_on],
+                  expire_on: values[:expire_on],
+                  started_at: values[:started_at],
+                  ended_at: values[:ended_at],
+                  time_to_live: values[:time_to_live],
+                  message_id: values[:message_id],
+                  process_status: create_process_status,
+                  transmittable_errors: [],
+                  allow_list: [],
+                  deny_list: []
+                })
+      rescue StandardError => e
+        Rails.logger.error { "Couldn't create job #{e.backtrace}" }
+      end
+
+      def create_job_entity(job_hash)
+        validation_result = AcaEntities::Protocols::Transmittable::Operations::Jobs::Create.new.call(job_hash)
+
+        validation_result.success? ? Success(validation_result.value!) : Failure("Unable to create job due to invalid params")
+      end
+
+      def generate_job_id(key)
+        "#{key}_#{Time.now.to_i}"
+      end
+
+      def create_process_status
```

Comment:
Consider making this a monadic step as this method could potentially return a failure monad. Also, this applies to app/domain/operations/transmittable/create_transaction.rb and app/domain/operations/transmittable/create_transmission.rb

Reply:
Will do thanks for the suggestion.